File: AlarmClock/ios/AlarmClock/BackgroundAudioModule.m
//
//  BackgroundAudioModule.m
//  AlarmClock
//
//  Created by Claude Code
//  React Native bridge for BackgroundAudioManager
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(BackgroundAudioManager, NSObject)

// Start background audio session
RCT_EXTERN_METHOD(startBackgroundAudio:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

// Stop background audio session
RCT_EXTERN_METHOD(stopBackgroundAudio:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

// Check if background audio is running
RCT_EXTERN_METHOD(isBackgroundAudioRunning:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: AlarmClock/src/types/BackgroundAudioModule.ts
/**
 * BackgroundAudioModule.ts
 * TypeScript types for the native BackgroundAudioManager module
 */

export interface BackgroundAudioResponse {
  status: string;
  message: string;
}

export interface BackgroundAudioStatusResponse {
  isRunning: boolean;
}

export interface BackgroundAudioModule {
  /**
   * Start the background audio session with silent sound loop
   * This allows the app to maintain an active audio session when backgrounded
   */
  startBackgroundAudio(): Promise<BackgroundAudioResponse>;

  /**
   * Stop the background audio session
   * Call this when no alarms are scheduled or app is being terminated
   */
  stopBackgroundAudio(): Promise<BackgroundAudioResponse>;

  /**
   * Check if background audio is currently running
   */
  isBackgroundAudioRunning(): Promise<BackgroundAudioStatusResponse>;
}
